newsAggregator/controllers/authController.js

```diff
@@ -0,0 +1,78 @@
+const bcrypt = require("bcrypt");
+const path = require("path");
+const fs = require("fs");
+const jwt = require("jsonwebtoken");
+const Validator = require("../helpers/validator");
+
+const signUp = (req, res) => {
+  const { name, email, password, preferences } = req.body;
+  const userDataPath = path.join(__dirname, "../src", "users.json");
+  const userData = fs.readFileSync(userDataPath, {
+    encoding: "utf8",
+    flag: "r",
+  });
```

Comment:
Do try experimenting with the in memory store. Reading always from a file makes the app slow, as you are loading all the data of the file in memory. Eventually we will be using databases but for now do keep this in mind.